### ISSUE TITLE
Encryption at rest accepts any key

### ISSUE THREAD

**Issue by odbuser2** [orientechnologies/orientdb#6229]
## Expected behavior and actual behavior
Expected : Create followed by Open of an ODatabaseDocumentTx honors the encryption key.
Actual : Create followed by Open of an ODatabaseDocumentTx does not honor the encryption key during the Open.  The open ignores the encryption key in this case.

## Steps to reproduce the problem
Execute the deleteCreateOpenDeleteDB test which will show that you can create the database with an encryption key and then open it with an incorrect key in the same process.

To prove that the database is encrypted run the following tests independently in sequence:
Run the following test:
`deleteCreateOpenDeleteDB()`

This succeeds but it should not.  It creates the database with one key and then opens it with another key which should not be allowed.

To prove that the database is actually encrypted and requires the correct key to open, run the following one at a time (separate vm processes):
```
dropDB : drop the database if it exists
createDB : creates the db with the correct key
openDB : tries to open the db with an incorrect key and this correctly fails!
```
```
package orientdb;

import com.orientechnologies.orient.core.config.OGlobalConfiguration;
import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
import org.junit.Test;

public class EncryptionTest {
    private final String url    = "plocal:tmp/orientdb/encryption";
    private final String method = "aes";
    private final String key    = "E95F22239273CA4F52A8847F27F76507";
    private final String badKey = "11111111111111111111111111111111";
    private final String user   = "admin";
    private final String pword  = "admin";
    
    @Test
    public void deleteCreateOpenDeleteDB() {        
        dropDB();
        createDB();
        openDB();
        dropDB();
    }
    @Test
    public void createDB() {
        System.out.format("Creating %s%n", url);
        ODatabaseDocumentTx db = new ODatabaseDocumentTx(url);
        db.setProperty(OGlobalConfiguration.STORAGE_ENCRYPTION_METHOD.getKey(), method);
        db.setProperty(OGlobalConfiguration.STORAGE_ENCRYPTION_KEY.getKey(), key);
        db.create().close();
    }
    @Test
    public void openDB() {
        System.out.format("Opening %s%n", url);
        ODatabaseDocumentTx db = new ODatabaseDocumentTx(url);
        db.setProperty(OGlobalConfiguration.STORAGE_ENCRYPTION_KEY.getKey(), badKey);
        db.open(user, pword).close();
    }
    @Test
    public void dropDB() {
        System.out.format("Dropping %s%n", url);
        ODatabaseDocumentTx db = new ODatabaseDocumentTx(url);
        db.setProperty(OGlobalConfiguration.STORAGE_ENCRYPTION_KEY.getKey(), key);
        if (db.exists()) {
            db.open(user, pword).drop();
        }
    }
}
```

### Runninng Mode
- [x] Embedded, using PLOCAL access mode
- [ ] Embedded, using MEMORY access mode
- [ ] Remote

### Misc
- [ ] I have a distributed setup with multiple servers. How many? 
- [ ] I'm using the Enterprise Edition

### OrientDB Version
- [ ] v2.0.x - Please specify last number: 
- [ ] v2.1.x - Please specify last number: 
- [x] v2.2.x - Please specify last number: 0

### Operating System
- [ ] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

### Java Version
- [ ] 6
- [ ] 7
- [x] 8

**Comment by odbuser2** [comment]
Is this going to be in 2.2.4?

**Comment by SDIPro** [comment]
This works fine in 2.2.4.

**Comment by odbuser2** [comment]
v. 2.2.4
I see the same behavior as in my test.  I can still create a database with a key and then open immediately afterwards with a bad key and it succeeds.

**Comment by SDIPro** [comment]
Fixed in 2.2.5.

**Comment by odbuser2** [comment]
Thank you.  I verified that this is fixed in 2.2.6.

The exception is a little confusing when you try to open the db with the wrong key:
com.orientechnologies.orient.core.exception.OInvalidStorageEncryptionKeyException: Cannot decrypt content
Caused by: javax.crypto.BadPaddingException: Given final block not properly padded

I tried with a valid AES 256 bit key (but not the correct one to open the database) and this exception makes it appear as though the key itself is formatted incorrectly rather than the wrong key to open the database.
